soc/arm/nuvoton_npcx/common/scfg.c

```diff
@@ -26,15 +26,22 @@ struct npcx_scfg_config {
  * Get io list which default functionality are not IOs. Then switch them to
  * GPIO in pin-mux init function.
  *
- * def_io_conf: def-io-conf-list {
- *               compatible = "nuvoton,npcx-pinctrl-def";
+ * def-io-conf-list {
```

Comment:
Can you update the property name `pinctrl-0` to `pinxmux`?   This better matches what you storing into this property.

I also wonder if the `def-io-conf-list` can be converted to use the `pinctrl-0` and `pinctrl-names` properties?

Reply:
>Can you update the property name pinctrl-0 to pinxmux? This better matches what you storing into this property.

Done. Please notice a change [CL 3686727](https://chromium-review.googlesource.com/c/chromiumos/platform/ec/+/3686727) in the downstream repo is also necessary. 

>I also wonder if the def-io-conf-list can be converted to use the pinctrl-0 and pinctrl-names properties?

The type of `pinmux` property in pinctrl rebuild node is `phandle` not `phandles`. Zephyr device-tree macros cannot expand child nodes in a two-layer expansion if each element length of a property is various. For example:
```
pinctrl-0 = <&node1 &node2>;

node1 {
    pinmux = <&alt1 &alt2>;
};

node2 {
    pinmux = <&alt3 &alt4 &alt5>;
};
```
We will encounter a build error since the length of `pinmux` element is not fixed. I don't suggest applying pinctrl-0 and pinctrl-names mechanism here so far.